cf-execd/cf-execd-runner.c

```diff
@@ -609,6 +619,7 @@ static void MailResult(const ExecConfig *config, const char *file)
         if (CompareResultEqualOrFiltered(config, file, prev_file))
         {
             Log(LOG_LEVEL_VERBOSE, "Mail report: previous output is the same as current so do not mail it");
+            fclose(fp);
```

Comment:
This function is pretty huge, and you have to add fclose in many places. I've seen many cases of people forgetting to add a close or free in cases like this. A nice pattern is to put most of the code in a helper function, and then fclose after the function call. Keep in mind for the future.